packages/devextreme/js/__internal/scheduler/__tests__/views.test.ts

```diff
@@ -37,4 +37,39 @@ describe('views', () => {
     const appointment = container.querySelector('.dx-scheduler-appointment');
     expect(appointment !== null).toBe(true);
   });
+
+  it('should render all-day appointment correctly with fractional cell values', async () => {
+    const FRACTIONAL_CELL_WIDTH = 250.4;
+    setupSchedulerTestEnvironment({ width: FRACTIONAL_CELL_WIDTH });
+    const { container, scheduler } = await createScheduler({
+      dataSource: [{
+        text: 'Appointment 2',
+        startDate: new Date('2021-02-02T15:15:00.000Z'),
+        endDate: new Date('2021-02-02T17:45:00.000Z'),
+        allDay: true,
+        resourceId: 1,
+      }],
+      currentDate: new Date(2021, 1, 2),
+      startDayHour: 8,
+      endDayHour: 20,
+      allDayPanelMode: 'hidden',
+      groups: ['resourceId'],
+      resources: [{
+        label: 'User',
+        dataSource: [
+          { title: 'Mark', id: 1 },
+          { title: 'Luke', id: 2 },
+        ],
+        displayExpr: 'title',
+        fieldExpr: 'resourceId',
+        valueExpr: 'id',
+      }],
+    });
+
+    scheduler.option('width', 800);
+    scheduler.option('width', 730);
```

Comment:
lets remove width changes because it doesn't effect on the view in jest. We mock cell width at the start of the test